core-web/libs/sdk/client/src/lib/client/content/builders/collection/collection.ts

```diff
@@ -0,0 +1,334 @@
+            this.#rawQuery = buildQuery;
+
+            return this;
+        }
+
+        if (typeof buildQuery !== 'function') {
+            throw new Error('Parameter for query method should be a function or a string');
+        }
+
+        const builtQuery = buildQuery(new QueryBuilder());
+
+        // This can be use in Javascript so we cannot rely on the type checking
+        if (builtQuery instanceof Equals) {
+            this.#query = builtQuery.raw(this.currentQuery.build());
+        } else {
+            throw new Error('Provided query is not valid.');
+        }
+
+        return this;
+    }
+
+    /**
+     * Takes a boolean that represents if the content to query should be on draft or not
+     *
+     * Note: The default value is false to fetch content that is not on draft
+     *
+     * @param {boolean} draft A boolean that represents the state of the content
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    draft(draft: boolean): this {
+        this.#query = this.currentQuery.field('live').equals((!draft).toString());
+
+        return this;
+    }
+
+    /**
+     * Takes a string that represents a variant ID of content created with the A/B Testing feature
+     *
+     * Note: variantId defaults to "DEFAULT" to fetch content that is not part of an A/B test
+     *
+     * @param {string} variantId A string that represents a variant ID
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    variant(variantId: string): this {
+        this.#query = this.currentQuery.field('variant').equals(variantId);
+
+        return this;
+    }
+
+    /**
+     * Takes a number that represents the depth of the relationships of a content
+     *
+     * Note: The depth is set to 0 by default and the max supported value is 3
+     *
+     * @param {number} depth The depth of the relationships of a content
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    depth(depth: number): this {
+        if (depth < 0 || depth > 3) {
+            throw new Error('Depth must be between 0 and 3');
+        }
+
+        this.#depth = depth;
+
+        return this;
+    }
+
+    /**
+     * Executes the fetch and returns a promise that resolves to the content or rejects to an error
+     *
+     * @param {OnFullfilled} [onfulfilled] A callback that is called when the fetch is successful
+     * @param {OnRejected} [onrejected] A callback that is called when the fetch fails
+     * @return {Promise<GetCollectionResponse<T> | GetCollectionError>} A promise that resolves to the content or rejects to an error
+     * @memberof CollectionBuilder
+     */
+    then(
+        onfulfilled?: OnFullfilled<T>,
+        onrejected?: OnRejected
+    ): Promise<GetCollectionResponse<T> | GetCollectionError> {
```

Comment:
If I use try/catch

```typescript
try {
  const data = await client.getCollection()...
} catch(e) {
  console.log(e)
}
```

And it failed, does the catch gets triggered?

Reply:
It will. I added a test for it